packages/local-build-plugin/src/artifacts.ts

```diff
@@ -35,10 +44,21 @@ export async function prepareArtifacts(artifactPaths: string[], logger?: bunyan)
     suffix = '.tar.gz';
     localPath = archivePath;
   }
-  const artifactName = `build-${Date.now()}${suffix}`;
+  const artifactName =
+    artifactType === ManagedArtifactType.APPLICATION_ARCHIVE
+      ? `build-${Date.now()}${suffix}`
+      : artifactType === ManagedArtifactType.BUILD_ARTIFACTS
+        ? `artifacts-${Date.now()}${suffix}`
+        : `xcode-logs-${Date.now()}${suffix}`;
   const destPath = config.artifactPath ?? path.join(config.artifactsDir, artifactName);
   await fs.copy(localPath, destPath);
-  l?.info({ phase: 'PREPARE_ARTIFACTS' }, `Writing artifacts to ${destPath}`);
+  if (artifactType === ManagedArtifactType.APPLICATION_ARCHIVE) {
+    l?.info({ phase: 'PREPARE_ARTIFACTS' }, `Writing application archive to ${destPath}`);
+  } else if (artifactType === ManagedArtifactType.BUILD_ARTIFACTS) {
+    l?.info({ phase: 'PREPARE_ARTIFACTS' }, `Writing build artifacts to ${destPath}`);
+  } else {
```

Comment:
Prefer switch over enums. This way if we add more managed types we won't forget to add handling.